app/join/check-email/route.ts

```diff
@@ -0,0 +1,13 @@
+import { NextResponse, type NextRequest } from "next/server";
+
+import { getUserProfileByMail } from "~/src/data/user-profile";
+import { looksLikeEmailAdress } from "~/src/utils";
+
+export async function GET(request: NextRequest): Promise<Response> {
+  const email = request.nextUrl.searchParams.get("email");
+  if (!email || typeof email !== "string" || !looksLikeEmailAdress(email)) {
+    return new Response("Required params missing or invalid.", { status: 400 });
+  }
+  const user = await getUserProfileByMail(email);
```

Comment:
Tady by to chtělo case-insensitive porovnání. Když zadám při registraci svůj e-mail, kterej obsahuje velká písmena, jenom lowercase, nenajde ho to. Ale registrace pak samozřejmě nelze dokončit, protože e-mail už existuje.

Případně můžeš asi rovnou použít funkci `getUserByEmail`, kde už máš e-mail normalizovaný?

Reply:
Jajky, pravda. Já celkově nejsem spokojen s tím, jak s tou normalizací pracujeme. Ona je nezbytně potřeba, aby lidi nenaráželi na blbé chyby kvůli mezeře nebo velkému písmenu navíc, když jsou e-maily case insensitive – ale asi bychom neměli spoléhat na to, že si každý volající bude tu adresu normalizovat sám. Ještě to projdu a zkusím to udělat tak, aby normalizace probíhala už v té funkci `getUserProfileByMail`. A přidám poznámku do dokumentace.